aptos-move/block-executor/src/executor.rs

```diff
@@ -1108,9 +1100,19 @@ where
                         &final_results,
                     ) {
                         if scheduler.halt() {
-                            let mut shared_commit_state_guard = shared_commit_state.acquire();
-                            let (_, maybe_error) = shared_commit_state_guard.dereference_mut();
-                            *maybe_error = Some(Error::FallbackToSequential(e));
+                            // Only one thread / worker will successfully halt, hence below
+                            // ExplicitSyncWrapper acquires are safe.
+
+                            if let BlockExecutionError::FatalVMError((inner_err, txn_idx)) = &e {
```

Comment:
what happens if error is to fallback to sequential?

Reply:
this error is specifically VM unrecoverable abort, and it has its own enum type (avoiding the fallback types)